extended/src/main/java/io/kubernetes/client/extended/kubectl/KubectlRolloutHistory.java

```diff
@@ -0,0 +1,161 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package io.kubernetes.client.extended.kubectl;
+
+import io.kubernetes.client.common.KubernetesObject;
+import io.kubernetes.client.extended.kubectl.exception.KubectlException;
+import io.kubernetes.client.extended.kubectl.util.deployment.DeploymentHelper;
+import io.kubernetes.client.openapi.ApiException;
+import io.kubernetes.client.openapi.apis.AppsV1Api;
+import io.kubernetes.client.openapi.models.V1Deployment;
+import io.kubernetes.client.openapi.models.V1ObjectMeta;
+import io.kubernetes.client.openapi.models.V1PodTemplateSpec;
+import io.kubernetes.client.openapi.models.V1ReplicaSet;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class KubectlRolloutHistory<ApiType extends KubernetesObject>
+    extends Kubectl.ResourceBuilder<ApiType, KubectlRolloutHistory<ApiType>>
+    implements Kubectl.Executable<ApiType> {
+
+  public static final String CHANGE_CAUSE_ANNOTATION = "kubernetes.io/change-cause";
+
+  public V1PodTemplateSpec getTemplate() {
+    return template;
+  }
+
+  public List<History> getHistories() {
+    return histories;
+  }
+
+  public static class History {
+    private final long revision;
+    private final String changeCause;
+
+    public long getRevision() {
+      return revision;
+    }
+
+    public String getChangeCause() {
+      return changeCause;
+    }
+
+    public History(long revision, String changeCause) {
+      this.revision = revision;
+      this.changeCause = changeCause;
+    }
+
+    @Override
+    public String toString() {
+      return "{revision :" + revision + ", changeCause :" + changeCause + "}";
+    }
+  }
+
+  private long revision;
+
+  private final List<History> histories;
+
+  private V1PodTemplateSpec template;
+
+  @Override
+  public ApiType execute() throws KubectlException {
+    validate();
+    AppsV1Api api = new AppsV1Api(this.apiClient);
+    try {
+      if (apiTypeClass.equals(V1Deployment.class)) {
+        V1Deployment deployment = api.readNamespacedDeployment(name, namespace, null, null, null);
+        deploymentViewHistory(deployment, api);
+        return (ApiType) deployment;
+      } else {
+        throw new KubectlException("Unsupported class for rollout history: " + apiTypeClass);
+      }
+    } catch (ApiException ex) {
+      throw new KubectlException(ex);
+    }
+  }
+
+  KubectlRolloutHistory(Class<ApiType> apiTypeClass) {
+    super(apiTypeClass);
+    revision = 0;
+    histories = new ArrayList<>();
+  }
+
+  public KubectlRolloutHistory<ApiType> revision(int revision) {
+    this.revision = revision;
+    return this;
+  }
+
+  private void validate() throws KubectlException {
+    StringBuilder msg = new StringBuilder();
+    if (name == null) {
+      msg.append("Missing name, ");
+    }
+    if (namespace == null) {
+      msg.append("Missing namespace, ");
```

Comment:
i can see it's a bit fiddling to deal with the capitalization conditionally here, personally i prefer to make them all lowercased simply